digdag-client/src/main/java/io/digdag/client/DigdagClient.java

```diff
@@ -569,14 +583,64 @@ public InputStream getLogFile(long attemptId, RestLogFileHandle handle)
 
     public InputStream getLogFile(long attemptId, String fileName)
     {
-        Response res = target("/api/logs/{id}/files/{fileName}")
-            .resolveTemplate("id", attemptId)
-            .resolveTemplate("fileName", fileName)
-            .request()
-            .headers(headers.get())
-            .get();
-        // TODO check status code
-        return res.readEntity(InputStream.class);
+        Invocation request = target("/api/logs/{id}/files/{fileName}")
+                .resolveTemplate("id", attemptId)
+                .resolveTemplate("fileName", fileName)
+                .request()
+                .headers(this.headers.get())
+                .buildGet();
+
+        return invokeWithRetry(request)
+                .readEntity(InputStream.class);
+    }
+
+    private Response invokeWithRetry(Invocation request)
+    {
+        Retryer<Response> retryer = RetryerBuilder.<Response>newBuilder()
+                .retryIfException(not(DigdagClient::isDeterministicError))
+                .withWaitStrategy(exponentialWait())
+                .withStopStrategy(stopAfterAttempt(10))
+                .build();
+
+        try {
+            return retryer.call(() -> {
+                Response res = request.invoke();
+                if (res.getStatusInfo().getFamily() != SUCCESSFUL) {
+                    res.close();
+                    return handleErrorStatus(res);
+                }
+                return res;
+            });
+        }
+        catch (ExecutionException | RetryException e) {
+            Throwable cause = e.getCause() != null ? e.getCause() : e;
```

Comment:
Hmm.... It looks RetryException includes the last exception not the first one. The first exception's information would be more helpful...